src/cdk/a11y/focus-trap/configurable-focus-trap-config.ts

```diff
@@ -15,5 +15,8 @@ export class ConfigurableFocusTrapConfig {
    * done manually by the user. Default is to create them
    * automatically.
    */
-  defer: boolean = false;
+  defer?: boolean = false;
+
+  /** Predicate function that determines whether the focus trap will allow focus to escape. */
+  focusEscapePredicate?: (target: HTMLElement) => boolean;
```

Comment:
Name bikesheding: WDYT about `allowFocusEscape`?

Reply:
I'm not too attached to the name, but I wanted to have "predicate" in there since we do the same in a few other places. To me, `allowFocusEscape` sounds like something you'd call on the focus trap to allow focus to escape, e.g. `focusTrap.allowFocusEscape()`.